interp/interp_test.go

```diff
@@ -556,6 +556,10 @@ var runTests = []runTest{
 		`a='"\n'; printf "%s %s" "${a}" "${a@E}"`,
 		"\"\\n \"\n",
 	},
+	{
+		"declare a; a+=(b); echo ${a[@]}",
```

Comment:
add `{#a[@]}` too; I'm not sure if the code should result in 1 or 2 array elements, and if we got it wrong in the code we wouldn't notice.

Reply:
And it's a real issue in this PR. Using prev.Str on an unset makes the length two (or if we fix #672). I'll special case this conversion since it's pretty common to declare and then add arrays in a loop.